app/yamlheaderchecker.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 Aetf <7437103@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <QRegularExpression>
+#include "yamlheaderchecker.h"
+
+YamlHeaderChecker::YamlHeaderChecker(const QString &text)
+{
+    QRegularExpression rx(QStringLiteral(R"(^---\s*\n(.*?\n)?(---|\.\.\.)\s*(\n|$))"),
+                          QRegularExpression::DotMatchesEverythingOption);
```

Comment:
Would you be interested to write a unit test for this class? It would be great if the unit test shows that this RegEx works as expected with different inputs.